rag_experiment_accelerator/doc_loader/documentIntelligenceLoader.py

```diff
@@ -44,35 +45,33 @@ def is_supported_by_document_intelligence(format: str) -> bool:
 
 def load_with_azure_document_intelligence(
     environment: Environment,
+    index_config: IndexConfig,
     file_paths: list[str],
-    chunk_size: int,
-    overlap_size: int,
-    azure_document_intelligence_model: str,
     **kwargs: dict,
 ) -> list[Document]:
     """
     Load pdf files from a folder using Azure Document Intelligence.
 
     Args:
-        environment (Environment): The environment class
+        environment (Environment): The environment class.
+        index_config (IndexConfig): The index configuration class.
         file_paths (list[str]): Sequence of paths to load.
-        chunk_size (int): The size of each text chunk in characters.
-        overlap_size (int): The size of the overlap between text chunks in characters.
-        azure_document_intelligence_model (str): The model to use for Azure Document Intelligence.
         **kwargs (dict): Unused.
 
     Returns:
         list[Document]: A list of Document objects.
     """
     documents = []
-    logger.info(f"Using model {azure_document_intelligence_model}")
+    logger.info(
+        f"Using model {index_config.chunking.azure_document_intelligence_model}"
+    )
     for file_path in file_paths:
```

Comment:
Severity: Major - The logger statement now references `index_config.chunking.azure_document_intelligence_model`, which improves clarity. Ensure that `index_config` is validated to prevent runtime errors if the attribute is missing.